generator/lib/src/field_element_ex.dart

```diff
@@ -134,12 +133,12 @@ extension FieldElementEx on FieldElement {
 
       // Validate indexes
       if ((((primaryKey ?? indexed) != null) && !(type.realmType?.mapping.indexable ?? false)) || //
-          (primaryKey != null && type.isDartCoreBool)) {
+          (primaryKey != null && (type.isDartCoreBool || type.isExactly<RealmValue>()))) {
```

Comment:
Lets add tests in the generator for @Indexed and @PrimaryKey annotations on a RealmValue member.

Reply:
Tests added